### ISSUE TITLE
[BUG] My Courses Map Error Message

### ISSUE THREAD

**Issue by vburgett** [rokwire/illinois-app#4314]
If a student with one online course at the U of I goes to Map > My Courses, they get the below error message. 

When an app user has no in-person courses or no courses at all, the pop-up window should say "There are no in-person courses to display."

iPhone 13
V6.0.36

Cc: @clint156 

![image](https://github.com/user-attachments/assets/2c5c087c-3310-4dfd-9c8a-c9fb537137b0)

**Comment by mihail-varbanov** [comment]
I am unable to recreate this issue. The message from the above screenshot is displayed then and only then if the HTTP request to courses/studentcourses API from Gateway BB fails for some reason. Please note that as far as I know Gateway BB requests in turn some other University services so the reasons for the failure could be outside the Rokwire ecosystem. This could be investigated in the server logs why this particular request failed.

When the request succeeds but there are no courses associated to the current user the application displays a different message:
![Screenshot 2024-08-26 at 3 33 45 PM](https://github.com/user-attachments/assets/0be0c4ae-fad4-46cf-896b-e0406b9bf5a3)

When the request succeeds and there is only one course in the response from Gateway BB the application displays its location on the map:
![Screenshot 2024-08-26 at 3 30 02 PM](https://github.com/user-attachments/assets/d755b5d8-004f-4d74-a964-3fff3efc7993)

**Comment by vburgett** [comment]
@cdukeUX do you see this error?

**Comment by cdukeUX** [comment]
Yes - I see "Failed to load student courses." One thing to note is that my course is an online course, so it would not have a physical location. @vburgett

**Comment by vburgett** [comment]
This error is still an issue in 7.3.17.

**Comment by clint156** [comment]
@mihail-varbanov On production, the building block is returning this if there are no courses found in the campus data:
HttpStatus = 404,
Content = {
    "message": "Error finding result for id",
    "status": "not-found"
}

It looks like in the My Courses panel, if the status is anything but a 200 it simply says something about not having any registered courses for the semester. What is the maps panel looking for when it tries to load the courses? The status? An empty list of courses?  I can change the building block if that is the direction you want to go.

**Comment by mihail-varbanov** [comment]
Hi @clint156,

In Map, if the request succeeds and we get an empty list of courses as a result we report `No student courses registered`, if the request fails and we get a null list of courses we report `Failed to load student courses`.

In the My Courses panel, we report `You do not appear to be enrolled in any courses for the selected term.` in the first case, and `You do not appear to be registered for any courses for the selected term.` in the second one (that I think is a little misleading).

As far as I understand the discussion so far, the problem is in the message text content displayed when the Summer term is selected? If we need to treat 404 response code as a successful request with an empty list of courses result, so that not to report that the request has failed, I can handle this in the client without any change on the backend. @vburgett, is this what you actually want?

**Comment by vburgett** [comment]
@mihail-varbanov yes, please. 

If there are no courses or only online courses, the message should say:

"You do not appear to be registered for any in-person courses for the selected term."

Or the shorter version:

"You do not appear to be registered for any in-person courses."

The other Map message should match too instead of saying "No student courses registered"

**Comment by mihail-varbanov** [comment]
@vburgett, this is done in version 7.3.30.
